airflow/jobs/scheduler_job.py

```diff
@@ -327,6 +327,13 @@ def _executable_task_instances_to_queued(self, max_tis: int, session: Session =
                 self.log.warning("Tasks using non-existent pool '%s' will not be scheduled", pool)
                 continue
 
+            pool_total = pools[pool]["total"]
+            if task_instance.pool_slots > pool_total:
+                self.log.warning(
+                    "Tasks requesting more slots than pool configured. pool '%s' will not be scheduled", pool
+                )
```

Comment:
```suggestion
                self.log.warning(
                    "Tasks requesting more slots (%s) than are configured in the pool. pool '%s' will not be scheduled", pool
                )
```

"pool '%s' will not be scheduled"" is not true though -- if there is another task that uses than pool and needs less slots it will go through

Reply:
improved the message in https://github.com/apache/airflow/pull/20178/commits/a5bee3a80ee03fb9b2a4aec79a6c50c8201a4d57